examples/Misc/SetCustomClientTimeouts.php

```diff
@@ -147,24 +163,35 @@ private static function serverStreamingCall(GoogleAdsClient $googleAdsClient, in
      * @param GoogleAdsClient $googleAdsClient the Google Ads API client
      * @param int $customerId the customer ID
      */
-    private static function unaryCall(GoogleAdsClient $googleAdsClient, int $customerId)
+    private static function makeUnaryCall(GoogleAdsClient $googleAdsClient, int $customerId)
     {
         $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
         // Creates a query that retrieves all campaign IDs.
         $query = 'SELECT campaign.id FROM campaign';
 
         $output = '';
         try {
-            // Issues a search request.
+            // Issues a search request by setting a custom client timeout.
             $response = $googleAdsServiceClient->search(
                 $customerId,
                 $query,
                 [
-                    'timeoutMillis' => self::CUSTOM_CLIENT_TIMEOUT,
+                    // As of V5, any unary call is retryable and has default retry settings.
+                    // Complete information about these settings can be found here:
+                    // https://googleapis.github.io/gax-php/master/Google/ApiCore/RetrySettings.html.
+                    // For this particular call, the default retry settings can be found in the
+                    // following file:
+                    // https://github.com/googleads/google-ads-php/blob/master/src/Google/Ads/GoogleAds/V5/Services/resources/google_ads_service_client_config.json.
+                    //
+                    // When making an unary call, an optional argument is provided and can be
+                    // used to override the default retry settings with given values.
                     'retrySettings' => [
-                        'initialRpcTimeoutMillis' => self::CUSTOM_CLIENT_TIMEOUT,
-                        'maxRpcTimeoutMillis' => self::CUSTOM_CLIENT_TIMEOUT,
-                        'totalTimeoutMillis' => self::CUSTOM_CLIENT_TIMEOUT
+                        // Sets the timeout that is used for the first try.
+                        'initialRpcTimeoutMillis' => self::CLIENT_TIMEOUT_MILLIS,
+                        // Sets the maximum timeout that can be used for any given try.
+                        'maxRpcTimeoutMillis' => self::CLIENT_TIMEOUT_MILLIS,
+                        // Sets the maximum accumulative timeout of the call, it includes all tries.
```

Comment:
If I guess correctly, once the call fails, it will use the initialRpcTimeoutMillis option to calculate the next timeout to wait, which is supposed to be larger than initialRpcTimeoutMillis, right?

If so, would it make more sense to set the three options differently?

Reply:
There is no one-fit-all configuration. Here, I use the same as what we generate by default but with a custom value (see the `3600000` occurences in this [example](https://github.com/googleads/google-ads-php/blob/master/src/Google/Ads/GoogleAds/V2/Services/resources/ad_service_client_config.json#L16)). We set `initialRpcTimeoutMillis` and `maxRpcTimeoutMillis` to the maximum timeout because we want to maximize the timeout used by the tries from the really first try, this is to maximize our chances of success. If a try fails and there is some time left, the next try will use a timeout that will be lower than `initialRpcTimeoutMillis` because capped by the accumulative timeout `maxRpcTimeoutMillis`.

Another option would be to disable retries overriding the setting `retriesEnabled` for the call. This would be less confusing because only one setting `timeoutMillis` would then be used like for `GAS.SearchStream`. But it will be less resilient because it will not retry if a transient error occurs during the call (unrelated to timeout).

What do you think?

Reply:
I didn't realize that the three params were also set to be the same by default.
I'm fine with it, although I just think that if we can at least separate `maxRpcTimeoutMillis` from `total_timeout_millis`, it will provide us venue to explain the relationships between them for beginners.

As an super simplified example, if `maxRpcTimeoutMillis` is 1 and `total_timeout_millis` is 5, you can explain that in general, this will retry around 5 times until it reaches 5 millis, which is `total_timeout_millis`.

Reply:
Sounds good, I edited to reflect that.